benedict/dicts/keylist/keylist_dict.py

```diff
@@ -72,7 +72,12 @@ def _get_by_keys(self, keys, default=None):
         parent, key, _ = keylist_util.get_item(self, keys)
         if type_util.is_list(parent) and type_util.is_wildcard(key):
             return parent
-        if type_util.is_dict(parent):
+        elif type_util.is_wildcard(keys[-2]):
+            if type_util.is_list_of_dicts(parent):
+                return [item.get(key) for item in parent]
+            if type_util.is_list_of_list(parent):
+                return _
```

Comment:
There is not final `else`, what happens if `type_util.is_wildcard(keys[-2])` and `parent` is a list of items of different type, eg. `list`, `dict`, `int`, `bool` and `str` at the same level or it is not a `list` at all?

Reply:
Can you build a test case? I am trying to figure out the one that confirm the necessity of implementing this. Because those statements covers "complex" data structure. eg. list of int cannot be access by ```"a[*].1"``` and it should throw ```KeyError``` , so "grandparent" wildcard grandparent shouldn't be possible for list of scalar values.

Reply:
So we can assume that the wildcard will work only if the list is a list of `dict`|`list`, otherwise it raises a `KeyError`?

If this is the behaviour for me it's ok.